lib/improver/tests/metadata/test_check_datatypes.py

```diff
@@ -0,0 +1,307 @@
+            enforce.check_cube_not_float64(self.cube)
+
+    def test_float64_cube_coord_points_with_fix(self):
+        """Test a failure of a cube with 64 bit coord points."""
+        self.cube.coord("projection_x_coordinate").points = (
+            self.cube.coord("projection_x_coordinate").points.astype(
+                np.float64))
+        expected_cube = self.cube.copy()
+        expected_cube.coord("projection_x_coordinate").points = (
+            expected_cube.coord("projection_x_coordinate").points.astype(
+                np.float64))
+        expected_coord = expected_cube.coord("projection_x_coordinate")
+        enforce.check_cube_not_float64(self.cube, fix=True)
+        self.assertEqual(self.cube, expected_cube)
+        self.assertEqual(
+            self.cube.coord("projection_x_coordinate"), expected_coord)
+
+    def test_float64_cube_coord_bounds(self):
+        """Test a failure of a cube with 64 bit coord bounds."""
+        x_coord = self.cube.coord("projection_x_coordinate")
+        # Default np.array for float input is np.float64.
+        x_coord.bounds = (
+            np.array([(point - 10., point + 10.) for point in x_coord.points])
+        )
+        msg = "64 bit coord bounds not allowed"
+        with self.assertRaisesRegex(TypeError, msg):
+            enforce.check_cube_not_float64(self.cube)
+
+    def test_float64_cube_coord_bounds_with_fix(self):
+        """Test a failure of a cube with 64 bit coord bounds."""
+        x_coord = self.cube.coord("projection_x_coordinate")
+        # Default np.array for float input is np.float64.
+        x_coord.bounds = (
+            np.array([(point - 10., point + 10.) for point in x_coord.points])
+        )
+        expected_cube = self.cube.copy()
+        expected_cube.coord("projection_x_coordinate").points = (
+            expected_cube.coord("projection_x_coordinate").points.astype(
+                np.float64))
+        expected_coord = expected_cube.coord("projection_x_coordinate")
+        enforce.check_cube_not_float64(self.cube, fix=True)
+        self.assertEqual(self.cube, expected_cube)
+        self.assertEqual(
+            self.cube.coord("projection_x_coordinate"), expected_coord)
+
+    def test_float64_cube_time_coord_points_ok(self):
+        """Test a pass of a cube with 64 bit time coord points."""
+        self.cube.coord("time").points = (
+             self.cube.coord("time").points.astype(np.float64))
+        enforce.check_cube_not_float64(self.cube)
+
+    def test_float64_cube_forecast_ref_time_coord_points_ok(self):
+        """Test a pass of a cube with 64 bit fcast ref time coord points."""
+        frt_coord = iris.coords.AuxCoord(
+            [np.float64(min(self.cube.coord("time").points))],
+            standard_name="forecast_reference_time")
+        self.cube.add_aux_coord(frt_coord)
+        enforce.check_cube_not_float64(self.cube)
+
+
+class Test_check_time_coordinate_metadata(IrisTest):
```

Comment:
It wouldn't hurt to add one that raised multiple errors. Could also add one where a coordinate isn't found.